### DISCUSSION TITLE
Basic auth in swagger end points for downstream server

### DISCUSSION THREAD

**Discussion by Amardeep08** [Burgyn/MMLib.SwaggerForOcelot#215] [Q&A]
Anyway to enforce basic authentication to swagger endpoints for downstream servers? 
Swagger endpoints are protected with basic auth.
Achieved this in ocelot with delegating handler..
But in swagger...unable to do it?

**Comment by Burgyn** [comment]
Hi,
I'm sorry but I don't quite understand your question.

Could you please be more specific? Can you please describe the more detailed behavior you expect. Ideal if you make a simple example to demonstrate this.

Thanks.

**Reply by Amardeep08** [reply]
Swagger UI endpoints is getting 401 error.
So, I created delegating handler and implemented   basic auth with help of httpclient and able to do basic auth and call api endpoints in postMan.
OnSendAsync(context)
{ httpclient.sendAsync(requestbasic,cancella6 token)}.

But for swagger, it returns  500 internal error . In error message, it is failing due to invalid authentication . The reason ,as per my understanding, the application is not able to retrieve swagger.json from swagger endpoints .
So, I am trying to find a way to add authentication before just hitting swagger endpoints.

**Reply by Burgyn** [reply]
Hi,

the only thing I can think of is to use `DownstreamSwaggerHeaders` option.

```csharp
app.UseSwaggerForOcelotUI(opt => {
  opt.DownstreamSwaggerHeaders = new[]
  {
      new KeyValuePair<string, string>("Auth-Key", "AuthValue"),
  };
})
```

These headers will be used when requesting the downstream swagger endpoint.

**Reply by jariahh** [reply]
@Burgyn I'm trying to get it to pass along my bearer token while making the request for the json. As each of my downstream services will use the token to provide the correct json back based on your authentication level. How can I achieve this. The downstream Swagger Headers looked promising, but I think It would be` new KeyValuePair<string, string>("Authorization", "Bearer {dynamic token goes here}")`

**Reply by Burgyn** [reply]
Hi @jariahh 
What exactly you want to authorize? Accessing swagger json or your downstream resource API endpoints?

**Reply by jariahh** [reply]
Hey @Burgyn the downstream end points are already working with authorization, just the actual call to get the swagger json doesn't appear to be getting the bearer token.

**Reply by Burgyn** [reply]
Hi @jariahh,

try looking at this https://github.com/Burgyn/MMLib.SwaggerForOcelot/issues/83
